bin/anthology/events.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019-2020 Marcel Bollmann <marcel@bollmann.me>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import defaultdict
+from copy import deepcopy
+from slugify import slugify
+from typing import List
+import logging as log
+import os
+import re
+import yaml
+
+try:
+    from yaml import CLoader as Loader
+except ImportError:
+    from yaml import Loader
+
+from .utils import is_newstyle_id, build_anthology_id, deconstruct_anthology_id
+from anthology.data import VENUE_FORMAT
+
+
+class EventIndex:
```

Comment:
add comment for what it does: "Keeps track of all events in the anthology and their relation to venues and volumes"?